npm/test/clientfileprocessing.test.ts

```diff
@@ -261,6 +261,78 @@ test("file successfully uploaded to s3", async () => {
   expect(s3UploadMock.uploadToS3).toHaveBeenCalledTimes(1)
 })
 
+test("s3ProgressBarCallback function updates the progress bar with the percentage processed", () => {
+  setupUploadPageHTML()
+
+  const progressInformation: IProgressInformation = {
+    totalFiles: 1,
+    percentageProcessed: 100,
```

Comment:
I would test with a percentage other than 100 here. You test this at the end, and with this test we are a checking the weighting calculation is working as expected. Would say try 75 percent.